features/support/hooks.ts

```diff
@@ -28,7 +28,7 @@ Before(function (
   this.tmpDir = path.join(
     projectPath,
     'tmp',
-    `${path.basename(pickle.uri)}_${line.toString()}`
+    `${path.basename(pickle.uri) ?? ''}_${(line ?? '').toString()}`
```

Comment:
What's the need for this change?

Reply:
A new eslint rule that bubbled up

Reply:
I don't understand how that is possible. If you revert this does the build fail?

Reply:
We've recently upgraded the linter, which has new rules. Rules are only validated against modified files (that's how husky is set up).

So the linter only kicked in now that @davidjgoss modified the file.

Perhaps we should configure husky to lint/format everything? It will make commits a bit slower, but we avoid these kinds of surprised.

Reply:
What is husky? Can you point me to the commit that introduced that? I personally prefer to lint everything each build and think linting modified files only is only useful if you are starting linting and don't want to fix everything at once (though there are also other ways of dealing with that too, ie disable the rules with errors and enable each separately when fixing them all)

Reply:
Never mind my last comment - we're not using husky.

[Husky](https://github.com/typicode/husky) is a tool that runs arbitrary commands on git commit. It's popular for linting before commit.

Since we don't have this, what I think happened is:

* eslint was upgraded
* we don't enforce linting of all source code on commit
* a file that's not respecting the new eslint rules was updated
* the linter kicked in, forcing an unrelated change to be added just to keep the linter happy

Reply:
AFAIK we lint all files new and old - previously had to fix some new errors to help dependabot do a eslint version bump for example. And the only change in this file is the one I made to fix the lint issue.

I can't quite figure out what has happened, but I wasn't getting the error locally, then saw it on Travis, then `rm -r`'d my `node_modules` and did a fresh yarn install, and then was able to see the error. This should be impossible because of lockfiles.

Reply:
It's curious, but I don't think it's worthwhile spending more time understanding how this happened.

Is anything holding up merging this?

Reply:
Agree could be a rabbit hole. FWIW I do think it's a bit of an over-the-top rule.

Reply:
Can you please revert this change? I'd like to see the commit this error occur on CI? If it is still happening can we fix this in another PR? I'd like to narrow the discussion around it as I don't think both changes are needed

Reply:
Reverted - and of course now it looks to be passing.

Here's [the build where it failed](https://travis-ci.org/github/cucumber/cucumber-js/builds/727824227) on this rule so I don't look totally mad.